pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,434 @@
+            });
+});
+
+class BusWorkflowRow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        let label = this.props.workflow[1];
+        const match = /(?:Report to) (.*)/.exec(this.props.workflow[1]);
+        if (match !== null) {
+            label = match[1];
+        }
+
+        this.state = {
+            label: label,
+            message: "",
+            problemState: ProblemState.REPORTABLE,
+            reportLinks: [],
+            task: null,
+        };
+
+        this.onCancelButtonClick = this.onCancelButtonClick.bind(this);
+        this.onReportButtonClick = this.onReportButtonClick.bind(this);
+        this.updateStatusFromBus = this.updateStatusFromBus.bind(this);
+
+        this.updateStatusFromBus();
+    }
+
+    onCancelButtonClick(event) {
+        this.state.task.Cancel();
+    }
+
+    onReportButtonClick(event) {
+        if (this.state.problemState === ProblemState.REPORTING) {
+            return;
+        }
+
+        this.setState({
+            message: "Waiting to start…",
+            problemState: ProblemState.REPORTING,
+        });
+
+        reportd_client.promise().then((client) => {
+            client.call("/org/freedesktop/reportd/Service",
+                        "org.freedesktop.reportd.Service", "CreateTask",
+                        [this.props.workflow[0], this.props.problem.path])
+                    .done((object_path) => {
+                        const task_proxy = client.proxy("org.freedesktop.reportd.Task", object_path);
+
+                        task_proxy.wait().done((object_path) => {
+                            task_proxy.addEventListener("changed", (event, data) => {
+                                switch (data.Status) {
+                                case TaskState.CANCELED:
+                                    this.setState({ message: "Reporting was canceled", });
+                                    // falls through
+                                case TaskState.ERROR:
+                                    this.setState({ problemState: ProblemState.REPORTABLE, });
+                                    break;
+                                case TaskState.COMPLETED:
+                                    this.setState({ problemState: ProblemState.REPORTED, });
+                                    break;
+                                default:
+                                    break;
+                                }
+
+                                this.updateStatusFromBus();
+                            });
+                            task_proxy.addEventListener("Prompt", (event, object_path, message, type) => {
+                                this.setState({ message: "Waiting for input…" });
+                                const task_prompt = client.proxy("org.freedesktop.reportd.Task.Prompt", object_path);
+                                const props = {
+                                    body: message,
```

Comment:
It should not be string but element.

Reply:
How come? I can’t see anything that would suggest that.

Reply:
https://github.com/cockpit-project/cockpit/blob/master/pkg/lib/cockpit-components-dialog.jsx#L297

Reply:
You’ll have to talk to me like I’m five. AFAIK, pure text is completely fine in DOM. If you look at how the property is used above, that certainly appears legal.

Reply:
It is legal for sure and it for sure works. But its type is defined as React element and you pass it String. So it is just for a good measure.
More about typechecking of props can be found at https://reactjs.org/docs/typechecking-with-proptypes.html